CorsixTH/Lua/research_department.lua

```diff
@@ -478,6 +478,15 @@ function ResearchDepartment:improveMachine(machine)
     research_info.start_strength = research_info.start_strength
       + improve_rate
     research_info.strength_imp = research_info.strength_imp + 1
+    -- Update dynamic info for any machines that now have improved strength.
+    for _, room in ipairs(self.world.rooms) do
+      for obj in pairs(room.objects) do
+        if(obj.object_type.id == machine.id) then
+          obj:updateDynamicInfo(false)
+          break
```

Comment:
Your `break` here ends the for loop, but what if there's multiple rooms that need this update? Maybe we have 2 tongue slicers, but here it would only update one.

Reply:
I've tested this on a hospital with multiple rooms of the same type. The break only breaks from the existing loop - i.e. no need to look for further objects in that room and move on to the next room (in the outer for loop).

Reply:
Oh yeas, ignore me :P